### PR TITLE
feat(swap): Add Changelly's Terms of Use in the confirmation screen [LIVE-1196]

### PR DESCRIPTION
**Pull request by Justkant**
Added terms of use for any provider in the swap config.

### Type

Feature

### Context

[LIVE-1196]

### Parts of the app affected / Test plan

- Swap flow

[LIVE-1196]: https://ledgerhq.atlassian.net/browse/LIVE-1196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by LFBarreto** [comment]
@Justkant i leave this in HODL mode cause v3 is right arround the corner to be merged it will required a rebase once its done

**Comment by Justkant** [comment]
> @Justkant i leave this in HODL mode cause v3 is right arround the corner to be merged it will required a rebase once its done

I'll be there to rebase or modify when needed

**Comment by juan-cortes** [comment]
Since v3 has already been merged, feel free to rebase/modify now if needed :trollface:

File: src/components/TermsFooter.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    return null;
+  }
+
+  return (
+    <>
+      <LText style={[styles.text]}>
+        {t("DeviceAction.confirmSwap.acceptTerms")}
+      </LText>
+      <LText style={[styles.text, styles.link]} onPress={onLinkClick}>
+        {t("DeviceAction.confirmSwap.termsLink", { provider })}
```
Comment (juan-cortes) [inline src/components/TermsFooter.js:27]: In order to capitalize the provider (which is something product was super worried about) you can do something like  we did for the `transfer.swap.form.summary.disclaimer.desc` wording. Basically something like
```
<Trans
  i18nKey={"DeviceAction.confirmSwap.termsLink"}
  values={{ provider }}
>
  <LText style={styles.capitalize} color="smoke" />
</Trans>
```

And the wording changing to 
`<0>{provider}}</0>'s terms of use`

**Review by juan-cortes — CHANGES_REQUESTED** [review]

File: src/components/TermsFooter.js (hunk 1 above)
Comment (Justkant) [reply src/components/TermsFooter.js:27]: Perfect thanks for the example, I knew this syntax with i18next but I didn't think of it

File: src/components/DeviceAction/index.js [hunk 2]
```diff
@@ -171,7 +171,15 @@ export default function DeviceAction<R, H, P>({
   }
 
   if (initSwapRequested && !initSwapResult && !initSwapError) {
-    return renderConfirmSwap({ t, device: selectedDevice, colors, theme });
+    // $FlowFixMe
```
[the remark this answers is not in the document]
Comment (Justkant) [reply src/components/DeviceAction/index.js]: I had an error before but just tried to remove it and it's gone, was probably just my editor

File: src/components/TermsFooter.js (hunk 1 above)
Comment (Justkant) [reply src/components/TermsFooter.js]: Done

File: src/components/DeviceAction/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Justkant) [reply src/components/DeviceAction/index.js]: Screenshot 2022-05-05 at 12 09 35

Probably why I had this, and I also see that every other use of the `request` var in this file adds a `// $FlowFixMe`

[the remark this answers is not in the document]
Comment (Justkant) [reply src/components/DeviceAction/index.js]: I need to check every variable to remove all flow errors on the generic ([more info](https://flow.org/en/docs/types/generics/#toc-adding-types-to-generics))
Screenshot 2022-05-05 at 12 25 25

Edit: just found out that I can also define the expect type for the generic
Screenshot 2022-05-05 at 12 28 13

**Review by juan-cortes — APPROVED** [review]

**Pull request closed without merging**